stdlib/Distributed/src/macros.jl

```diff
@@ -198,7 +202,8 @@ macro everywhere(procs, ex)
     imps = extract_imports(ex)
     return quote
         $(isempty(imps) ? nothing : Expr(:toplevel, imps...)) # run imports locally first
-        let ex = $(Expr(:quote, ex)), procs = $(esc(procs))
+        let ex = Expr(:toplevel, :(task_local_storage()[:SOURCE_PATH] = $(get(task_local_storage(), :SOURCE_PATH, nothing))), $(Expr(:quote, ex))),
```

Comment:
doesn't this need to nest and restore the old value?

Reply:
We know that `remotecall_eval` will evaluate the expression in a new task.